File: 2021/day-21/1.js
const numberOfRollsPerPlayer = 3
const boardSize = 10
let bigDie = 0
let totalRolls = 0

let player1 = 5
let player2 = 6

while (condition()) {
    player1 += (roll() + player1 % boardSize) % boardSize;
    if (!condition()) break;
    player2 += (roll() + player2 % boardSize) % boardSize;
}

console.log(player1, player2, totalRolls);

function condition() {
    return player1 < 1000 && player2 < 1000;
}

function die() {
    if (bigDie > 100) {
        bigDie = 0;
    }
}

function roll() {
    total = 0
    for (let i = 0; i < numberOfRollsPerPlayer; i++) {
        die();
        bigDie++;
        total += bigDie;
        totalRolls++;
    }
    return total;
}

File: 2021/day-21/1.py
bigDie = 1
numberOfRollsPerPlayer = 3
boardSize = 10
totalRolls = 0

print("Welcome to day 21 of Advent of Code")

player1 = 5
player2 = 6

def roll():
    total = 0
    for x in range(numberOfRollsPerPlayer):
        if bigDie > 100:
            bigDie = 1
        total += bigDie
        bigDie += 1
        totalRolls += 1
    return total % boardSize

while (player1 < 1000 and player2 < 1000):
    player1 += roll() 
    player2 += roll() 

print(player1, player2, totalRolls)
